src/server/server-preboot.module.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  InjectionToken,
+  ModuleWithProviders,
+  NgModule,
+  RendererFactory2,
+  RendererType2,
+  ViewEncapsulation,
+  APP_BOOTSTRAP_LISTENER
+} from '@angular/core';
+import { PlatformState } from '@angular/platform-server';
+import { PrebootOptions } from '../common';
+import { generatePrebootEventRecorderCode } from './preboot.generator';
+
+// only thing this does is modify the document
+@NgModule()
+export class ServerPrebootModule {
+
+  // user can override the default preboot options by passing them in here
+  static recordEvents(opts?: PrebootOptions): ModuleWithProviders {
+    return {
+      ngModule: ServerPrebootModule,
+      providers: [{
+
+        // this likely will never be injected but need something to run the
+        // factory function
+        provide: APP_BOOTSTRAP_LISTENER,
+
+        // generate the inline preboot code and inject it into the document
+        useFactory: function (state: PlatformState, rendererFactory: RendererFactory2) {
+          return function () {
+            const doc = state.getDocument();
+            const prebootEventRecorderCode = generatePrebootEventRecorderCode(opts);
+            addInlineCodeToDocument(prebootEventRecorderCode, doc, rendererFactory);
+          };
```

Comment:
On the server side we alter the document to stick the generated preboot code in there.